src/http/client.rs

```diff
@@ -283,9 +283,19 @@ impl EspHttpConnection {
     pub fn read(&mut self, buf: &mut [u8]) -> Result<usize, EspError> {
         self.assert_response();
 
-        Self::check(unsafe {
-            esp_http_client_read_response(self.raw_client, buf.as_mut_ptr() as _, buf.len() as _)
-        })
+        let result = Self::check(unsafe {
+            esp_http_client_read(self.raw_client, buf.as_mut_ptr() as _, buf.len() as _)
+        });
+
+        // workaround since esp_http_client_read does not yet return EAGAIN error in ESP-IDF v4.
+        // in ESP-IDF v5 esp_http_client_read will return EAGAIN and this should not be needed.
+        match result {
```

Comment:
I'm kind of OK with this approach, but isn't it better to actually branch the code for esp idf 5+ vs older esp idf-s and then do different things for each?

Reply:
It is all over the place but here you go:
`#[cfg(not(esp_idf_version_major = "4"))]`
^^^ For code which should run on 5+

and
`#[cfg(esp_idf_version_major = "4")]`
^^^ For code which should run on 4. We don't support ES:P IDF earlier than 4.3.

Reply:
With the above `cfg` hints written, I'm (as said earlier) kind of fine with the existing way too, so I can merge the current shape of the patch too.